plugins/package-managers/bazel/src/main/kotlin/Bazel.kt

```diff
@@ -163,11 +202,42 @@ class Bazel(
         return null
     }
 
-    private fun getPackages(scopes: Set<Scope>, registry: BazelModuleRegistryService): Set<Package> {
+    /**
+     * Get the packages for the dependencies of the given [scopes] using the given [registry], for the dependencies that
+     * don't have a local path override. The dependencies having an override defined in [localPathOverrides] are created
+     * without querying [registry] and will use [projectVcs] as VCS information.
+     */
+    private fun getPackages(
+        scopes: Set<Scope>,
+        registry: BazelModuleRegistryService,
+        localPathOverrides: Map<String, String>,
+        projectVcs: VcsInfo
+    ): Set<Package> {
         val ids = scopes.collectDependencies()
-        val moduleMetadataForId = ids.associateWith { getModuleMetadata(it, registry) }
-        val moduleSourceInfoForId = ids.associateWith { getModuleSourceInfo(it, registry) }
-        return ids.mapTo(mutableSetOf()) { getPackage(it, moduleMetadataForId[it], moduleSourceInfoForId[it]) }
+
+        val packagesWithPathOverrides = mutableSetOf<Package>()
+        val filteredIds = ids.filter { id ->
+            if (localPathOverrides.containsKey(id.name)) {
+                packagesWithPathOverrides += Package(
+                    id = id,
+                    declaredLicenses = emptySet(),
+                    description = "",
+                    homepageUrl = "",
+                    binaryArtifact = RemoteArtifact.EMPTY,
+                    sourceArtifact = RemoteArtifact.EMPTY,
+                    vcs = projectVcs.copy(path = localPathOverrides.getValue(id.name))
+                )
+                false
+            } else {
+                true
```

Comment:
This looks to me like a prime example for making use of `partition`, maybe like (not complete / untested):

        val (localIds, nonLocalIds) = ids.partition { it.name in localPathOverrides }

        val packagesWithPathOverrides = localIds.map {
            Package(
                id = it,
                declaredLicenses = emptySet(),
                description = "",
                homepageUrl = "",
                binaryArtifact = RemoteArtifact.EMPTY,
                sourceArtifact = RemoteArtifact.EMPTY,
                vcs = projectVcs.copy(path = localPathOverrides.getValue(id.name))
            )
        }

Reply:
I'll do it but maybe we will need another filtering when implementing https://github.com/oss-review-toolkit/ort/issues/8809.